evalml/automl/automl_search.py

```diff
@@ -778,10 +778,19 @@ def describe_pipeline(self, pipeline_id, return_dict=False):
         if pipeline_id not in self._results['pipeline_results']:
             raise PipelineNotFoundError("Pipeline not found")
 
+        automl_dict = dict()
+
         pipeline = self.get_pipeline(pipeline_id)
         pipeline_results = self._results['pipeline_results'][pipeline_id]
+        automl_dict.update(pipeline_results)
 
         pipeline.describe()
+
+        if pipeline.model_family == ModelFamily.ENSEMBLE:
+            input_pipeline_ids = [self._automl_algorithm._best_pipeline_info[model_family]["id"] for model_family in self._automl_algorithm._best_pipeline_info]
```

Comment:
If I'm understanding correctly, `_best_pipeline_info` will only contain one key per model family. So after the first ensemble pipeline is created, the data in `_best_pipeline_info` (may) be overwritten. This means that the `input_pipeline_ids` would always correspond to the latest ensemble and not necessarily the ensemble that was asked for.

Repro:

```python
from evalml.demos import load_breast_cancer
from evalml.automl import AutoMLSearch

X, y = load_breast_cancer()

# max_batches=19 means we ensemble twice
automl = AutoMLSearch(X, y, problem_type="binary", ensembling=True, max_batches=19)
automl.search()
automl.describe_pipeline(49)
```

We know the ids > 49 are wrong because the ensemble could have only been created with ids 1-48.

```
Input for ensembler are pipelines with IDs: [85, 32, 65, 53, 78, 16, 64]
```

Reply:
Amazing, thank you for catching this! 😅 

I updated the logic so that `self._results['pipeline_results']` stores the ids (using `_best_pipeline_info`) whenever an ensemble is built. I think this is fine, since when adding the ensemble, `_best_pipeline_info` has the right info for that pipeline (and as you pointed out, can later get overridden).

Test wise, updated `test_describe_pipeline_with_ensembling` to test for two ensembles and tested what you pointed out--that the ensemble should be created from ids less than the ensemble. Added a mock score that always decreases (latter pipelines do better) and then test that the second ensemble uses pipeline ids that come after the first ensemble pipeline.